pkg/util/ec2/ec2.go

```diff
@@ -96,14 +96,27 @@ func GetSourceName() string {
 var instanceIDFetcher = cachedfetch.Fetcher{
 	Name: "EC2 InstanceID",
 	Attempt: func(ctx context.Context) (interface{}, error) {
-		return getMetadataItemWithMaxLength(ctx, imdsInstanceID, false)
+		hostname, err := getMetadataItemWithMaxLength(ctx, imdsInstanceID, UseIMDSv2(false, false))
+		if err == nil && hostname != "" {
```

Comment:
We should not have to check both the error and the value (ie: the error should not be nil if we consider an empty hostname an error).

Reply:
done